controller/controller.go

```diff
@@ -53,175 +55,191 @@ func NewControllerWithClient(logger *log.Logger, client resources.StorageClient,
-func (c *Controller) Mount(mountRequest resources.FlexVolumeMountRequest) resources.FlexVolumeResponse {
+func (c *Controller) Mount(mountRequest k8sresources.FlexVolumeMountRequest) k8sresources.FlexVolumeResponse {
 	c.logger.Println("controller-mount-start")
 	defer c.logger.Println("controller-mount-end")
 	c.logger.Println(fmt.Sprintf("mountRequest [%#v]", mountRequest))
-
+	var lnPath string
 	attachRequest := resources.AttachRequest{Name: mountRequest.MountDevice, Host: getHost()}
 	mountedPath, err := c.Client.Attach(attachRequest)
 
 	if err != nil {
 		msg := fmt.Sprintf("Failed to mount volume [%s], Error: %#v", mountRequest.MountDevice, err)
 		c.logger.Println(msg)
-		return resources.FlexVolumeResponse{
+		return k8sresources.FlexVolumeResponse{
 			Status:  "Failure",
 			Message: msg,
-			Device:  "",
 		}
 	}
-	dir := filepath.Dir(mountRequest.MountPath)
-
-	c.logger.Printf("volume mounted at %s", mountedPath)
-	k8sRequiredMountPoint := path.Join(mountRequest.MountPath, mountRequest.MountDevice)
-	if _, err = os.Stat(k8sRequiredMountPoint); err != nil {
-		if os.IsNotExist(err) {
-
-			c.logger.Printf("creating volume directory %s", dir)
-			err = os.MkdirAll(dir, 0777)
-			if err != nil && !os.IsExist(err) {
-				msg := fmt.Sprintf("Failed creating volume directory %#v", err)
-				c.logger.Println(msg)
-
-				return resources.FlexVolumeResponse{
-					Status:  "Failure",
-					Message: msg,
-					Device:  "",
-				}
+	if mountRequest.Version == k8sresources.KubernetesVersion_1_5 {
+		dir := filepath.Dir(mountRequest.MountPath)
+		c.logger.Printf("mountrequest.MountPath %s", mountRequest.MountPath)
+		lnPath = mountRequest.MountPath
+		k8sRequiredMountPoint := path.Join(mountRequest.MountPath, mountRequest.MountDevice)
+		if _, err = os.Stat(k8sRequiredMountPoint); err != nil {
+			if os.IsNotExist(err) {
+
+				c.logger.Printf("creating volume directory %s", dir)
+				err = os.MkdirAll(dir, 0777)
+				if err != nil && !os.IsExist(err) {
+					msg := fmt.Sprintf("Failed creating volume directory %#v", err)
+					c.logger.Println(msg)
+
+					return k8sresources.FlexVolumeResponse{
+						Status:  "Failure",
+						Message: msg,
+					}
 
+				}
 			}
+		}
 
-			symLinkCommand := "/bin/ln"
-			args := []string{"-s", mountedPath, mountRequest.MountPath}
-			c.logger.Printf(fmt.Sprintf("creating slink from %s -> %s", mountedPath, mountRequest.MountPath))
-
-			cmd := exec.Command(symLinkCommand, args...)
-			_, err = cmd.Output()
-			if err != nil {
-				msg := fmt.Sprintf("Controller: mount failed to symlink %#v", err)
-				c.logger.Println(msg)
-				return resources.FlexVolumeResponse{
-					Status:  "Failure",
-					Message: msg,
-					Device:  "",
-				}
+	} else {
+		lnPath, _ = path.Split(mountRequest.MountPath)
+		c.logger.Printf("removing folder %s", mountRequest.MountPath)
 
-			}
-			msg := fmt.Sprintf("Volume mounted successfully to %s", mountedPath)
+		err = os.Remove(mountRequest.MountPath)
```

Comment:
please explain why u delete it?   add some doc string that explain the differenance because its not cleat from the code.
why in 1.5 we do create the directory